services/motion/builtin/builtin_test.go

```diff
@@ -2385,6 +2411,165 @@ func TestMoveCallInputs(t *testing.T) {
+
+	injectedVis := inject.NewVisionService("test-vision")
+	injectedCam := inject.NewCamera("test-camera")
+	injectedSlam := createInjectedSlam(
+		"test-slam",
+		"slam/example_cartographer_outputs/viam-office-02-22-3/pointcloud/pointcloud_4.pcd",
+		spatialmath.NewZeroPose(),
+	)
+
+	// set the dependencies
+	deps := resource.Dependencies{
+		fakeBase.Name():     fakeBase,
+		injectedVis.Name():  injectedVis,
+		injectedCam.Name():  injectedCam,
+		injectedSlam.Name(): injectedSlam,
+	}
+
+	// create service
+	ms, err := NewBuiltIn(ctx, deps, resource.Config{ConvertedAttributes: &Config{}}, logger)
+	test.That(t, err, test.ShouldBeNil)
+	t.Cleanup(func() { ms.Close(ctx) })
+
+	// create links for framesystem
+	baseLink := createBaseLink(t)
+	cameraGeom, err := spatialmath.NewBox(
+		spatialmath.NewZeroPose(),
+		r3.Vector{1, 1, 1}, "camera",
+	)
+	test.That(t, err, test.ShouldBeNil)
+	cameraLink := referenceframe.NewLinkInFrame(
+		baseLink.Name(),
+		spatialmath.NewPose(r3.Vector{0, 0, 0}, &spatialmath.OrientationVectorDegrees{OY: 1, Theta: -90}),
+		"test-camera",
+		cameraGeom,
+	)
+
+	// construct the framesystem
+	fsParts := []*referenceframe.FrameSystemPart{
+		{FrameConfig: baseLink},
+		{FrameConfig: cameraLink},
+	}
+	fsSvc, err := createFrameSystemService(ctx, deps, fsParts, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	// set the framesystem service for the motion service
+	ms.(*builtIn).fsService = fsSvc
+
+	// construct move request
+	moveReq := motion.MoveOnMapReq{
+		ComponentName: fakeBase.Name(),
+		Destination:   spatialmath.NewPoseFromPoint(r3.Vector{10, 0, 0}),
+		SlamName:      injectedSlam.Name(),
+		MotionCfg: &motion.MotionConfiguration{
+			PlanDeviationMM: 1,
+		},
+	}
+
+	planExecutor, err := ms.(*builtIn).newMoveOnMapRequest(ctx, moveReq, nil, 0)
+	test.That(t, err, test.ShouldBeNil)
+
+	mr, ok := planExecutor.(*moveRequest)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	// define injected method on vision service
+	injectedVis.GetObjectPointCloudsFunc = func(ctx context.Context, cameraName string, extra map[string]interface{}) ([]*viz.Object, error) {
+		boxGeom, err := spatialmath.NewBox(
+			spatialmath.NewPose(r3.Vector{4, 8, 10}, &spatialmath.OrientationVectorDegrees{OZ: 1}),
+			r3.Vector{2, 3, 5},
+			"test-box",
+		)
+		test.That(t, err, test.ShouldBeNil)
+		detection, err := viz.NewObjectWithLabel(pointcloud.New(), "test-box", boxGeom.ToProtobuf())
+		test.That(t, err, test.ShouldBeNil)
+		return []*viz.Object{detection}, nil
+	}
+
+	type testCase struct {
+		name          string
+		f             func(g spatialmath.Geometry) spatialmath.Geometry
+		detectionPose spatialmath.Pose
+	}
+	testCases := []testCase{
+		{
+			name:          "relative - SLAM/base theta does not matter",
+			f:             func(g spatialmath.Geometry) spatialmath.Geometry { return g },
+			detectionPose: spatialmath.NewPose(r3.Vector{4, 10, -8}, &spatialmath.OrientationVectorDegrees{OY: 1, Theta: -90}),
+		},
+		{
```

Comment:
we add an image with illustrations of the remains test cases to add intuition on the value of `detectionPose`
![IMG_0369](https://github.com/viamrobotics/rdk/assets/46872531/29ecb019-257d-49d0-b81c-14ec4c84735b)